src/Server.cpp

```diff
@@ -48,14 +48,23 @@ Server::Server(int port, const std::string &password)
 
 // const channelsMap &Server::getChannels() const { return _channels; }
 
-const Channel &Server::getChannelByName(const std::string &name) const {
+const Channel &Server::findChannelByName(const std::string &name) const {
   channelsMap::const_iterator it = _channels.find(name);
   if (it == _channels.end()) {
     throw std::runtime_error("Channel not found with the given name");
   }
   return it->second;
 }
 
```

Comment:
I may be better to rewrite like findClientByNickname() using a pointer to avoid the Server stops for non fatal reason (channel not found).
After change, we need check if the pointer is not NULL if it's not verified before.

```
Channel *Server::findChannelByName(const std::string &name) {
  channelsMap::iterator it = _channels.find(name);
  if (it == _channels.end())
   return (NULL);
  return (&it->second);
}
```